src/lib/platform/XWindowsScreen.cpp

```diff
@@ -829,35 +830,37 @@ XWindowsScreen::fakeMouseRelativeMove(SInt32 dx, SInt32 dy) const
 }
 
 void
-XWindowsScreen::fakeMouseWheel(SInt32, SInt32 yDelta) const
+XWindowsScreen::fakeMouseWheel(SInt32 xDelta, SInt32 yDelta) const
 {
-	// XXX -- support x-axis scrolling
-	if (yDelta == 0) {
-		return;
-	}
+    int numEvents;
 
-    int numEvents = accumulateMouseScroll(yDelta);
+    if ((!xDelta && !yDelta) || (xDelta && yDelta)) {
+        // Invalid scrolling inputs
+        return;
+    }
 
-    // choose button depending on rotation direction
-    const unsigned int xButton = mapButtonToX(static_cast<ButtonID>(
```

Comment:
Why this workaround got deleted?

Reply:
I don't believe this workaround was actually doing anything; the values passed to `mapButtonToX` was always either `-1` or `-2`, which got mapped to `4` and `5` respectively. I don't see how it could ever return `0` here, and thus go into the `if` block.

Reply:
Sounds reasonable then.

Reply:
I would have done this removal as a separate commit though, it would be much clearer to anyone who comes by it in the git history.